core/platform/email/mailgun_email_services.py

```diff
@@ -0,0 +1,38 @@
+# coding: utf-8
+#
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides mailgun api to send email."""
+import requests
+
+import feconf
+
+def send_mail(
+        sender_email, recipient_email, subject, plaintext_body,
+        html_body, bcc=None):
+    if not feconf.MAILGUN_DOMAIN_NAME:
```

Comment:
Yeah I think we can. I am yet to implement this and it will take some time. So once I am done with this I will push new commit. All your other suggestions have been marked and I have changed code / comments according to them.

Thanks